sdk/storage/azure-storage-blob/src/main/java/com/azure/storage/blob/BlobServiceAsyncClient.java

```diff
@@ -329,6 +331,72 @@ public String getAccountUrl() {
                 null, Context.NONE), timeout);
     }
 
+
+    /**
+     * Returns a reactive Publisher emitting the blobs in this account whose tags match the query expression. For more
+     * information, including information on the query syntax, see the <a href="DOC LINK NEEDED">Azure Docs</a>.
+     *
+     * <p><strong>Code Samples</strong></p>
+     *
+     * {@codesnippet com.azure.storage.blob.BlobServiceAsyncClient.filterBlobs#String}
+     *
+     * @param query Filters the results to return only blobs whose tags match the specified expression.
+     * @return A reactive response emitting the list of blobs.
+     */
+    public PagedFlux<FilterBlobItem> findBlobsByTags(String query) {
+        return this.findBlobsByTags(query, null, null);
+    }
+
+    /**
+     * Returns a reactive Publisher emitting the blobs in this account whose tags match the query expression. For more
+     * information, including information on the query syntax, see the <a href="DOC LINK NEEDED">Azure Docs</a>.
+     *
+     * <p><strong>Code Samples</strong></p>
+     *
+     * {@codesnippet com.azure.storage.blob.BlobAsyncServiceClient.filterBlobs#String-Integer}
+     *
+     * @param query Filters the results to return only blobs whose tags match the specified expression.
+     * @param maxResultsPerPage The maximum number of results to return in a given page.
+     * @return A reactive response emitting the list of blobs.
+     */
+    public PagedFlux<FilterBlobItem> findBlobsByTags(String query, Integer maxResultsPerPage) {
+        try {
+            return findBlobsByTags(query, maxResultsPerPage, null);
+        } catch (RuntimeException ex) {
+            return pagedFluxError(logger, ex);
+        }
+    }
+
+    // TODO: options instead of maxResults?
+    // TODO: Return type?
```

Comment:
could we return like BlobItem? What are the properties we want to expose on this?

Reply:
Unfortunately the results here only come back with container name, blob name, and tag key/value. Because BlobItem has so many properties, it seemed weird to return it when most of them would be null, also there's no place for the tag key/value to go. But it's also kinda lame that we have a whole different list result type for this. 

I don't really think there's a good alternative, but I left it open in case other people could think of one and because I can't